lib/sf-creating-dialog-view.coffee

```diff
@@ -0,0 +1,78 @@
+{TextEditorView, View, $$} = require 'atom-space-pen-views'
+
+module.exports =
+class SfCreatingDialogView extends View
+  @content: ->
+    @div class: 'sf-dialog-panel', =>
+      @label "Label"
+      @subview 'labelElement', new TextEditorView(mini: true)
+      @label "API Name"
+      @subview 'apiNameElement', new TextEditorView(mini: true)
+      @label "API Version"
+      @select class: "form-control", outlet: "apiVersionElement"
+      @button class: "btn btn-success create-btn disabled", outlet: "createButton"
+      @button class: "btn btn-danger cancel-btn", outlet: "cancelButton", "Cancel"
+
+  initialize: (dialog) ->
+    @model = dialog;
+
+    #Set initial values
+    @createButton.text("Create " + @model.itemType);
+    @labelElement.getModel().getBuffer().setText(@model.label);
+    @apiNameElement.getModel().getBuffer().setText(@model.apiName);
+
+    for apiVersionItem in @model.apiVersions
+      thisOption = $$ ->
+        @option value: apiVersionItem, apiVersionItem
+      if apiVersionItem == @model.apiVersion
+        thisOption.attr "selected", "selected"
+      @apiVersionElement.append(thisOption);
+
+    #Set methods
+    @labelElement.getModel().onDidChange () =>
+      if @model

```

Comment:
This checking for view provider - if we don't check this - we have some troubles when we change preferences for text editor